src/metabase/models/permissions.clj

```diff
@@ -1079,6 +1079,49 @@
     (when (and has-native-perms? (not (group-has-native-perms? group-or-id database-or-id)))
       (grant-native-readwrite-permissions! group-or-id database-or-id))))
 
+; Audit Permissions helper fns
+
+(defenterprise default-audit-db-id
+  "OSS implementation of `audit-db/default-audit-db-id`, which is an enterprise feature, so does nothing in the OSS
+  version."
+  metabase-enterprise.audit-db [] ::noop)
+
+(defenterprise default-audit-collection-entity-id
+  "OSS implementation of `audit-db/default-audit-collection-entity-id`, which is an enterprise feature, so does nothing in the OSS
+  version."
+  metabase-enterprise.audit-db [] [])
+
+(defn update-audit-collection-permissions
+  "Will remove or grant audit db (AppDB) permissions, if the instance analytics permissions changes."
+  [group-id changes]
+  (when config/ee-available?
+    (let [audit-collection-id  (-> (t2/select-one 'Collection :entity_id (default-audit-collection-entity-id))
+                                   :id)]
+      (doseq [[change-id type] changes]
+        (when (= change-id audit-collection-id)
+          (let [change-permissions! (case type
+                                      :read  grant-permissions!
+                                      :none  delete-related-permissions!
+                                      :write (throw (ex-info (tru
```

Comment:
So for making the instance analytics collection view only, the code will throw an error while checking for the instance analytics collection. I did it this way over checking the collection types because checking each collection for their type would require fetching the collection type of each changed collection id. If we have more read-only collection types, then I think it makes sense to do this, but for now I did it this way.